main/ejml-dsparse/src/org/ejml/sparse/csc/CommonOps_DSCC.java

```diff
@@ -1835,5 +1837,111 @@ public static double trace( DMatrixSparseCSC A ) {
 
         return output;
     }
+
+    /**
+     * This applies a given unary function on every value stored in the matrix
+     *
+     * @param input  (Input) input matrix. Not modified
+     * @param func   Unary function accepting a double
+     * @param output (Input/Output) Matrix. Modified.
+     */
+    public static void apply(DMatrixSparseCSC input, DoubleUnaryOperator func, @Nullable DMatrixSparseCSC output) {
+        if (output == null) {
+            output = input.createLike();
+        } else if (input != output) {
+            output.copyStructure(input);
+        }
+
+        for (int i = 0; i < input.nz_length; i++) {
+            output.nz_values[i] = func.apply(input.nz_values[i]);
+        }
+    }
+
+    public static void apply(DMatrixSparseCSC input, DoubleUnaryOperator func) {
+        apply(input, func, input);
+    }
+
+    /**
+     * This accumulates the matrix values to a scalar value
```

Comment:
Maybe add matlab like code to the description explaining what reduce is?

Reply:
I am not very familiar with matlab, but I will come up with something (probably one of the existing methods has suitable matlab-code, I can reuse)

Reply:
Doesn't have to be matlab just some concise summary. In a few places you can see pseudo code to describe what the function does.